src/ServicePulse.Host/app/js/services/services.monitoring.js

```diff
@@ -2,74 +2,65 @@
-                    endpoints = response.data;
-                }    
-            
-                var url = uri.join(scConfig.monitoring_url, '/data');
-                return $http.get(url)
-                    .then(function (response) {
-                        response.data["NServiceBus.Endpoints"].forEach(function(item) {
-                            var e = endpoints.filter(function (endpoint) { return endpoint.title === item.Name; });
-                            if (e.length === 1) {
-                                item.Errors = e[0].count;
-                            }
-                        });
-                        endpoints.filter(function(endpoint) {
-                            return response.data["NServiceBus.Endpoints"].filter(function(search) {
-                                    return search.Name === endpoint.title;
-                                }).length ===
-                                0;
-                        }).forEach(function(missingEndpoint) {
-                            response.data["NServiceBus.Endpoints"].push({
-                                "Name": missingEndpoint.title,
-                                "Errors": missingEndpoint.count,
-                                "Data": {
-                                    "Timestamps": [],
-                                    "CriticalTime": [],
-                                    "ProcessingTime": []
-                                }
-                            });
+        function getEndpoints() {
+            var mappedUrls = [{
+                url: uri.join(scConfig.service_control_url, 'recoverability', 'endpoints'),
+                mapper: function (endpoint) {
+                    return {
+                        Name: endpoint.title,
+                        Count: endpoint.count,
+                        IsFromSC: true
+                    };
+                }
+            }].concat(scConfig.monitoring_urls.map(function (url) {
+                    return {
+                        url: uri.join(url, '/data')
+                    };
+            }));
+
+            return rx.Observable.merge(mappedUrls.map(function (mappedUrl) {
+                var httpRequest = rx.Observable.just(mappedUrl.url)
+                    .flatMap(function (requestUrl) {
+                        var request = $http.get(requestUrl).then(function (result) {
+                            return mappedUrl.mapper ? result.data.map(mappedUrl.mapper) : result.data["NServiceBus.Endpoints"];
                             });
-                        return response;
-                    })
-                    .catch(function(fallback) {
-                        notifications.pushSticky('Can not connect to Monitoring Service');
-                    });
-            });
-            return outcome;
-        }
 
-        var previousExceptionGroupEtag;
+                        return request;
+                    }).retryWhen(automaticRetry);
```

Comment:
Given that the monitoring services could be running in different regions, I'd say it's a good thing to have. What are your specific concerns around it?